src/random_unitary.jl

```diff
@@ -0,0 +1,69 @@
+# Version of `sign` that returns one
+# if `x == 0`.
+function nonzero_sign(x)
+  iszero(x) && return one(x)
+  return sign(x)
+end
+
+using LinearAlgebra: LinearAlgebra, Diagonal, diag
+function qr_positive(M::AbstractMatrix)
```

Comment:
This is equivalent to `MatrixAlgebraKit.qr_full(!)(M; positive=true)`, so maybe we can immediately use that?

Reply:
Yes, ideally we would use that directly.